image.md

```diff
@@ -0,0 +1,140 @@
+
+For example, a listing of the `tar` of an image might look like this:
+```
+config.json
+rootfs/
+    bin/myapp
+requires.json
+sha512-1d62d181e9c1322d56ccd3a29d05018399147a16188dbd861c0279ad0dd7e14c/
+    config.json
+	rootfs/
+	    bin/runit
+	whiteout.json
+	ID
+sha512-291e2a171ef9bef8a838c59406d9b0aeb6f2f0ebe5173415205733d3d18b8e03/
+    config.json
+	rootfs/
+	    bin/monitor
+	whiteout.json
+	ID
+whiteout.json
+ID
+```
+
+## Dependent Images
+
+If an image has dependent images then it MUST include a `requires.json` file that references those images.
+The `requires.json` file MUST be a json encoded array matching this format:
+```
+{
+  "requires": [
+    {
+      "ID": "..."
+    },
+	{
+	  "ID": "..."
+	}, ...
+  ]
+}
+```
+
+The order of the images, `ID`s, in the array MUST be in the order in which the images are to be "layed down" into the filsystem bundle that is generated from this image.
+In other words, the first image in the array is written to the filesystem first, and subsequent images are overlayed on top - potentially overlaying an ealier image's files.
+
+QUESTION: what about deleted files? Do we want to mandate a whiteout scheme?
+
+## Image ID
+
+An image's `ID` MUST be a sha512 hash of image's contents in the order specified in the [Image Contents](#image-contents) section.
+
+An image's `ID` MUST be a string of the form `sha512-<hash>`.
+
+QUESTION: Do we really want to mandate sha512?
+
+## Whiteout List
+
+An image's `whiteout.json` file MUST be a JSON encoded file matching this format:
+```
+{
+  "files": [
+    "filename",
+	...
+  ]
+}
+```
+where each `filename` MUST be the absolute path to the file.
+
+## Expanding an Image to Create a Filesystem Bundle
+
+When an image is expanded to create a filesystem bundle the image's `config.json` file MUST be written to the root of the filesystem bundle's directory.
+The expansion process MUST create a directory for the container's root filesystem with a name as specified in the `config.json` file - ie. the `root/path` property.
+The contents of this directory MUST appear to be materizlied in such a way as to contain all of the files from the image, without the whiteout (deleted) files, and its dependent images.
+The exact mechanism, process and order, by which those files are materialized on disk is an implementation detail, however in the end it MUST appear as though all of the files from the images were written to disk in the order they were specified in the `requires.json` file.
+This means that each dependent image's `whiteout.json` file MUST appear to be processed during the materialization of that image so that if a subsequent image creates a file by the same name as one mentioned in the `whiteout.json` file then it will not be deleted.
+
+The `config.json` file of any dependent images are ignored by the expansion process.
```

Comment:
Then why require them?  If the image format is *just* a mechanism for constructing a rootfs from layers, I don't see how it improves on our existing mounts.